src/mesh/voxel_cloud.rs

```diff
@@ -1200,13 +1262,98 @@ impl ScalarField {
                     }
                 }
             }
-
-            // Process the current voxel. If it is outside the volumes, set its
-            // value to be positive, if it's inside, set it to negative.
-            self.voxels[one_dimensional] = if discovered_as_outer_and_empty[one_dimensional] {
-                Some(distance)
-            } else {
-                Some(-distance)
+            let is_outside = discovered_as_outer_and_empty[one_dimensional];
+
+            // Process the current voxel.
+
+            self.voxels[one_dimensional] = match falloff_function {
+                FalloffFunction::Linear(mul) => {
```

Comment:
The following should be used sparingly, but I think this is a good occasion for it:

Instead of having the logic here (and potentially duplicating it soon in the future), we could have `FalloffFunction::apply(&self, distance: f32) -> f32` and just call it here. You can add the apply associated function in a regular impl block for `FalloffFunction`.

Reply:
good, i like it. what about doing the same for `interpolate_to` and `add_values` and potentially other arithmetic operations on scalar fields?

Reply:
side note: we need to reveal the scalar fields in the UI. they are becoming beautifully powerful!

Reply:
> what about doing the same for interpolate_to and add_values and potentially other arithmetic operations on scalar fields?

`add_values` already has it (doesn't it?). We can add it to other scalar field methods, but since that isn't directly affecting metaballs, let's do it in a separate PR and for now just create an issue for that (or a fixme).